code-challenges.test.js

```diff
@@ -52,5 +109,30 @@ const numbersToAdd2 = [0, 7, -8, 12]
 const numbersToAdd3 = []
 // Expected output: []
 
+describe("addEm", () => {
+    it("takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.", () => {
+        expect(addEm(numbersToAdd1)).toEqual([2, 6, 51, 60]);
+        expect(addEm(numbersToAdd2)).toEqual([0, 7, -1, 11]);
+        expect(addEm(numbersToAdd3)).toEqual([])
+    })
+})
 
 // b) Create the function that makes the test pass.
+
+// This program will allow the user to take in an array and return an array that has the accumulating sum of the previous array. Empty array will return an empty array. 
+// Create a function expression. (addEm)
+// Create a variable that uses the map method on a parameter. 
+    // Within map, enter an element and index value parameter. The index will indicate what value the element starts at. The element will be the value of each element mapped, summed into a new array. 
+    // Return the variable of the new array.
+// Console log with the challenge parameters.
+
+let addEm = (a) => {
+    x = a.map((sum => value => sum+= value)(0))
+    return x
+}
+
+console.log(addEm(numbersToAdd1))
+
+console.log(addEm(numbersToAdd2))
+
+console.log(addEm(numbersToAdd3))
```

Comment:
Sam, what beautiful code!  Fantastic pseudocode - clear, detailed steps.  Your tests are solid and your code is clean.  I highly recommend using more descriptive variables.  I know it won't look as nice, but future you and other developers will have to research what each variable represents, which is no fun.  This is the only improvement I have for you as far as coding goes.  You're killing it!